src/NumericalAlgorithms/Spectral/Projection.hpp

```diff
@@ -14,20 +14,60 @@ class Mesh;
 
 namespace Spectral {
 
+/// The portion of an element covered by a child element.
+enum class ChildSize { Full, UpperHalf, LowerHalf };
+
 /// The portion of an element covered by a mortar.
-enum class MortarSize { Full, UpperHalf, LowerHalf };
+using MortarSize = ChildSize;
 
-std::ostream& operator<<(std::ostream& os, MortarSize mortar_size) noexcept;
+std::ostream& operator<<(std::ostream& os, ChildSize mortar_size) noexcept;
 
 /*!
- * \brief The projection matrix from a 1D mortar to an element.
+ * \brief The projection matrix from a child element to its parent.
  *
- * \details
  * The projection matrices returned by this function (and by
- * projection_matrix_element_to_mortar()) define orthogonal projection
- * operators between the spaces of functions on the boundary of the
- * element and the mortar.  These projections are usually the correct
- * way to transfer data onto and off of the mortars.
+ * projection_matrix_parent_to_child()) define orthogonal projection operators
+ * between the spaces of functions on a parent element and its children. These
+ * projections are usually the correct way to transfer data between elements in
+ * a mesh-refinement hierarchy, as well as between an element face and its
+ * adjacent mortars.
+ *
+ * These functions assume that the `child_mesh` is at least as fine as the
+ * `parent_mesh`, i.e. functions on the `parent_mesh` can be represented exactly
+ * on the `child_mesh`. In practice this means that functions can be projected
+ * to a mortar (the `child_mesh`) from both adjacent element faces (the
+ * `parent_mesh`) without losing accuracy. Similarly, functions in a
+ * mesh-refinement hierarchy don't lose accuracy when an element is split
+ * (h-refined). For this reason, the `projection_matrix_child_to_parent` is
+ * sometimes referred to as a "restriction operator" and the
+ * `projection_matrix_parent_to_child` as a "prolongation operator".
+ *
+ * The projections implemented here are \f$L^2\f$ projections, or "Galerkin"
+ * projections. They are defined such that the projection from a parent to one
+ * of its children ("prolongation") is an (exact) interpolation, and the
+ * projection from a child to its parent ("restriction") is its inverse.
+ * Specifically, for every function \f$u_f\f$ on the "fine" child-mesh \f$e_f\f$
+ * and every function \f$u_c\f$ on the "coarse" parent-mesh \$e_c\f$ the
+ * restriction operator \f$R\f$ satisfies the condition:
+ *
+ * \f{equation}
+ * \int_{e_c} R(u_f) u_c \mathrm{d}x = \int_{e_f} u_f I(u_c) \mathrm{d}x
```

Comment:
is `e` the element? We've been using the `\Omega` notation to denote an element, generally, I think.

I'm also not sure this is actually true on a curved mesh. The L2 projection on a curved mesh should be projecting `u J` instead of `u`. I think here you are assuming `J` is spatially constant.

Reply:
- I think `\Omega` usually denotes the full domain. For the elements I haven't seen a consistent notation. Some use `\Omega_i`, so should we do that as well? I like `e` because it's short and abbreviates "element" pretty clearly, but I don't feel strongly about it.
- The Jacobian from logical to inertial coords is in the `dx`, though I could make that explicit by writing `J d\xi` if you prefer.

Reply:
- at least the cases I've encountered `\Omega` has been the element, but sometimes used interchangeably with the domain. I'm fine with `\Omega_i` or `\Omega_k` as well. I don't love `e`
- This Jacobian is actually important because the L2 projection matrices _depend_ on the Jacobian. That is, the L2 matrices are defined as the integral of the basis functions with the Jacobian included. In order for projection to be conservative one needs to project `u J` using the "logical projection matrix" (this picks up some aliasing errors but is accurately to nearly roundoff), or one computes the (time-dependent) projection matrices `int l_i(xi) l_j(xi) J d \xi`. In other words, one must acknowledge that this Jacobian exists, either by saying 1) "We don't care about conservation", 2) "We take the aliasing hit and project `u J`", or 3) "We compute the (time-dependent) projection matrix on each element"
 To be clear, I'm not asking for a code change, I think 1 or 2 are the right thing to do, but it should be documented that projection and interpolation are _never_ equal on curved meshes but that the logical projection and interpolation matrices are equal when projecting to a higher-dimensional space (dimension in the sense of basis functions).

Reply:
I pushed a fixup that adds docs on Jacobians. Please let me know if the added docs make sense, or suggest an alternative wording.

One note to make sure I understand your comment above:
- The matrix `int l_i(xi) l_j(xi) J d \xi` is the _mass matrix_, not the projection matrix, though of course the projection (restriction) matrix `R = M_c^-1 I^T M_f` is computed from mass matrices so this might be what you meant? Note that R is implemented here in terms of Vandermonde matrices, but that's equivalent since M = (V V^T)^-1. Note also that this implementation in terms of Vandermonde matrices means that we're _not_ doing mass-lumping (diagonal mass-matrix approximation) here, as we do e.g. for lifting. That's probably fine, I just wanted to point it out. I added a bit on this to the docs as well.